lib/rules/indent.js

```diff
@@ -106,43 +108,51 @@ module.exports = function(context) {
          * @private
          */
         function getFixerFunction() {
-            var rangeToFix = [];
+            var replacement = "" + new Array(needed + 1).join(indentChar),  // replace with repeat in future
+                endOfPreviousToken = -1,
+                endOfLeadingComments = -1,
+                previousNewlineIndex = -1,
+                rangeToFix,
+                startOfLine,
+                referenceToken,
+                previousToken,
+                leadingComments;
+
+            if (isLastNodeCheck === true) {
+                referenceToken = sourceCode.getLastToken(node);
+            } else {
+                referenceToken = sourceCode.getFirstToken(node);
+            }
 
-            if (needed > gotten) {
-                var spaces = "" + new Array(needed - gotten + 1).join(indentChar);  // replace with repeat in future
+            // Three candidates for "start of line" for indent purposes.
 
-                if (isLastNodeCheck === true) {
-                    rangeToFix = [
-                        node.range[1] - 1,
-                        node.range[1] - 1
-                    ];
-                } else {
-                    rangeToFix = [
-                        node.range[0],
-                        node.range[0]
-                    ];
-                }
+            // 1. End of leading comment.
+            leadingComments = sourceCode.getComments(node).leading;
 
-                return function(fixer) {
-                    return fixer.insertTextBeforeRange(rangeToFix, spaces);
-                };
-            } else {
-                if (isLastNodeCheck === true) {
-                    rangeToFix = [
-                        node.range[1] - (gotten - needed) - 1,
-                        node.range[1] - 1
-                    ];
-                } else {
-                    rangeToFix = [
-                        node.range[0] - (gotten - needed),
-                        node.range[0]
-                    ];
-                }
+            if (leadingComments && leadingComments.length) {
+                // Use the end of the comment as the "start of line"
+                endOfLeadingComments = leadingComments[leadingComments.length - 1].range[1];
+            }
 
-                return function(fixer) {
-                    return fixer.removeRange(rangeToFix);
-                };
+            // 2. End of previous token.
+            previousToken = sourceCode.getTokenBefore(referenceToken);
```

Comment:
FYI do we have any tests that actually take advantage of this code? I removed the code completely and saw no change to the tests....